src/platforms/ctxlink/platform.c

```diff
@@ -67,26 +87,75 @@ static void adc_init(void)
 		__asm__("nop");
 }
 
+//
+// Use the passed string to configure the USB UART
+//
+// e.g. 38400,8,N,1
+bool platform_configure_uart(char *configuration_string)
+{
+	if (strlen(configuration_string) > 5) {
+		uint32_t baudRate;
+		uint32_t bits;
+		uint32_t stopBits;
+		char parity;
+		int count = sscanf(
+			configuration_string, "%" SCNd32 ",%" SCNd32 ",%c,%" SCNd32 "", &baudRate, &bits, &parity, &stopBits);
+		if (count == 4) {
+			uint32_t parityValue;
+			usart_set_baudrate(USBUSART, baudRate);
+			usart_set_databits(USBUSART, bits);
+			usart_set_stopbits(USBUSART, stopBits);
+			switch (parity) {
+			default:
+			case 'N': {
+				parityValue = USART_PARITY_NONE;
+				break;
+			}
+
+			case 'O': {
+				parityValue = USART_PARITY_ODD;
+				break;
+			}
+
+			case 'E': {
+				parityValue = USART_PARITY_EVEN;
+				break;
+			}
+			}
+			usart_set_parity(USBUSART, parityValue);
+		}
```

Comment:
This should probably have an `else` block that follows it, allowing this function to `return false;` when the incorrect number of things get parsed by sscanf(). It looks like that was the intent with the original logic save for `fResult` not being initialised when defined.

Reply:
Refactored the function to return false on failure.